play/src/main/scala/com/managementapp/authentication/controllers/AuthenticationController.scala

```diff
@@ -0,0 +1,32 @@
+package com.managementapp.authentication.controllers
+
+import play.api.mvc.{Action, AnyContent, Controller}
+import com.managementapp.common.actions.AuthenticatedAction
+import com.google.inject.Inject
+import com.managementapp.common.util.ProtobufUtil
+import com.managementapp.transfers.Authentication.AccountDTO
+import com.managementapp.authentication.services.AuthenticationService
+import com.managementapp.common.services.UserService
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class AuthenticationController @Inject() (authService: AuthenticationService, userService: UserService) extends Controller with ProtobufUtil {
+  def sampleActionThatRequiresAuth: Action[AnyContent] = AuthenticatedAction(authService) { request =>
+    //this route requires login first
+    Ok("Logged in user is: " + request.user.email)
+  }
+
+  def createAccount: Action[AnyContent] = Action {
+    request =>
+      val account = parseJSONProtocolBuffer[AccountDTO](AccountDTO.newBuilder, request)
```

Comment:
Protobuf is just a wrapper around json (or other messaging formats we use). Frontend and backend both need to have a protobuf parser/writer. The request content still has to be sent through HTTP JSON content type. 

So for example, we have this Account DTO(see Authentication.proto), because it is generated from protobuf, the protobuf parser can parse it to JSON and then send through HTTP. And then front-end receives the JSON, it STILL needs a protobuf parser to parse the JSON to the **Account DTO in Typescript** otherwise it cannot use it.

Protobuf becomes useful when we have complex logic later when dealing with Quizzes and other stuff. Because we can just define *.proto files and front-end and back-end can know exactly what is in the request, and what should be in the request. (And since we are using ts and scala we also enjoy the type-safety on json)

AccountDTO just happens to have only two simple fields. When DTO gets complex we will need proto files to organize our request content. So for consistency, all our DTOs should be defined through protobuf